### ISSUE TITLE
typescript support

### ISSUE THREAD

**Issue by Akimotorakiyu** [FEAScript/FEAScript#6]
typescript support

**Comment by nikoscham** [comment]
> typescript support

Thank you for the suggestions and for providing clear references on TypeScript advantages. I will investigate how can I gradually introduce TypeScript into FEAScript. I’ll update you and the community as I make progress on this.

**Comment by nikoscham** [comment]
> It seems that this project is still in its early stages, Is there any way I can help you? Testing, building, or other

Thank you so much for offering to help! From what I’ve seen of your projects, it looks like you have experience with library packaging. If that’s something you’d be interested in, we could collaborate on setting up the necessary configuration to publish the library as an npm package.

**Comment by sridhar-mani** [comment]
> First of all TypeScript is JavaScript with syntax for types.
> 
> > TypeScript adds additional syntax to JavaScript to support a tighter integration with your editor. Catch errors early in your editor. TypeScript understands JavaScript and uses type inference to give you great tooling without additional code.
> 
> There are too many reason and there are too many blog to tell you that what is TypeScript and why should you use it. I won't say much more. Simply put, typescript support is basically the basic consensus of front-end development.
> 
> You can click on the link below to take a look.
> 
> > https://www.google.com.hk/search?q=why+TypeScript&oq=why+TypeScript+&gs_lcrp=EgZjaHJvbWUyBggAEEUYOTIHCAEQABiABDIICAIQABgHGB4yCAgDEAAYBxgeMggIBBAAGAcYHjIICAUQABgHGB4yCAgGEAAYBxgeMggIBxAAGAcYHjIICAgQABgHGB4yBwgJEAAYgATSAQgyNTE4ajBqNKgCALACAQ&sourceid=chrome&ie=UTF-8
> 
> > https://stackoverflow.com/questions/12694530/what-is-typescript-and-why-should-i-use-it-instead-of-javascript
> 
> > https://www.contentful.com/blog/what-is-typescript-and-why-should-you-use-it/

Yup, I agree with that—TypeScript is great, especially for general-purpose libraries like UI components or frameworks. It helps with tooling, editor support, and catching errors early.

But when you're dealing with more complex libraries that mimic C++ applications—like CAD engines or scientific computation libraries—I don't think TypeScript is always necessary. In many cases, it's more about performance and low-level logic than developer ergonomics.

Also, libraries like OpenCascade or VTK.js often either don’t have full TypeScript support or use very basic typings. So unless you're planning to publish it as a reusable library where developer experience really matters, going full TypeScript might not bring much benefit.
